.github/PULL_REQUEST_TEMPLATE/pull_request_template.md

```diff
@@ -0,0 +1,55 @@
+- [ ] Add [CHANGELOG](https://github.com/timescale/timescaledb/blob/master/CHANGELOG.md) updates
+- [ ] Needs [documentation](https://github.com/timescale/docs) updates
+
+Fixes #<issue number>.
+
+---
+Cut everything below the above line.
+
+## Guidelines for Pull requests 
+
+Checklist for a pull request:
+
+- Try to follow the [the seven rules of a great Git commit
+message](https://chris.beams.io/posts/git-commit/).
+- Ideally, the PR should be merged as a single commit.
+- Rebase on latest master code.
+- Reference the issue(s) resolved with Fixes #<issue number>, or
+  Closes #<issue number>.
+- Make sure the PR has appropriate CHANGELOG updates, including thanks
+  to people that reported issues.
+- Include appropriate documentation changes.
+- Two approvals are necessary to merge.
+
+### The seven rules of a great Git commit message
+
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters
+3. Capitalize the subject line
+4. Do not end the subject line with a period
+5. Use the imperative mood in the subject line
+6. Wrap the body at 72 characters
+7. Use the body to explain what and why vs. how
+
+### Single commit PR
+
+The simplify reviewing and to protect against accidental merges of
+work-in-progress commits, the CI system enforces that a PR is merged
+as a single commit. However, sometimes a multi-commit PR is warranted
+if each commit is a distinct change that makes sense to submit in the
+same PR. The single-commit enforcement can be disabled by adding the
+following trailer:
+
+Disable-check: commit-count
+
+Generally, though, multi-commit PRs are discouraged as that either
+means the code change is too big, the PR mixes unrelated changes, or
+has a lot of work-in-progress commits that are not appropriate for a
+well-formatted commit log.
```

Comment:
I don't think there is agreement on this topic. The reason for the check is to prevent accidental merges where squash was forgotten.

Reply:
These are general guidelines. What are your arguments against it. The check is already there in CI and it can be disabled.

Reply:
Not sure if that is your intention but this makes it sound like there is something wrong with multi-commit PRs  and while most of our PRs are single-commit a PR with multiple commits is perfectly acceptable.

Reply:
It says it is discouraged, which in my understanding means multi-commit PRs can happen but we prefer single commit PRs. This is our current de facto policy and the vast majority of PRs are single commits already.

Further:

- I also think it is (should) be uncontroversial to say a PR should not mix unrelated changes. Please explain why you would want this, if you believe it is a good thing.
- We also don't want PRs to be too big (ideally) since that is a huge reviewer burden and it slows down the PR process. I don't think this should be controversial, but please explain why you would prefer big PRs if that is the case.
- And, we don't want to merge work-in-progress commits (which is the main reason we have the CI check).

There are of course exception to every rule, and we have a way to allow that already.

I can try to tweak the formulation. But please provide a concrete suggestion for improvement, because I don't think anything above _should_ be controversial our outside our current de facto policy.

Reply:
Tweaked the formulation.